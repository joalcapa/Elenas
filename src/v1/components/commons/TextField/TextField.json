{
  "description": "Este componente renderiza un texto basico",
  "displayName": "TextField",
  "methods": [],
  "props": {
    "text": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Texto del TextField",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "type": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Define el tama√±o del texto",
      "defaultValue": {
        "value": "'normal'",
        "computed": false
      }
    },
    "isTitle": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Define el grosor del texto",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    }
  }
}
