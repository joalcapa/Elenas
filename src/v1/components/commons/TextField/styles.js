import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
    title: {
        fontWeight: 'bold',
    },
});

export default styles;