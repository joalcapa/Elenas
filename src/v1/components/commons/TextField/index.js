import React from 'react';
import PropTypes from 'prop-types';
import {Text} from 'react-native';

import {FontSize} from '../../../resources';

import styles from './styles';

export const SMALL_TYPE = 'small';
export const NORMAL_TYPE = 'normal';
export const EXTRA_TYPE = 'extra';
export const BIG_TYPE = 'big';
export const HIGH_TYPE = 'high';

const TextField = (props) => (
    <Text style={[
        {fontSize: FontSize[props.type]},
        props.isTitle ? styles.title : {},
    ]}>{props.text}</Text>
);

TextField.propTypes = {
    text: PropTypes.string,
    type: PropTypes.string,
    isTitle: PropTypes.bool,
};

TextField.defaultProps = {
    text: '',
    type: NORMAL_TYPE,
    isTitle: false,
};

export default TextField;