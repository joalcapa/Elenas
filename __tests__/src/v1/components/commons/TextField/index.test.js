import React from 'react';
import {create} from 'react-test-renderer';

import TextField, {
    SMALL_TYPE,
    EXTRA_TYPE,
    BIG_TYPE,
    HIGH_TYPE,
} from '../../../../../../src/v1/components/commons/TextField';

describe('TextField component', () => {
    test('render empty text component', () => {
        const tree = create(<TextField />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render text component with text', () => {
        const tree = create(<TextField text={'Test text'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render text component with text and title', () => {
        const tree = create(<TextField isTitle text={'Test text'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render text component with text and title and fontSize of 14px', () => {
        const tree = create(<TextField isTitle text={'Test text'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render text component with text and title and fontSize of 12px', () => {
        const tree = create(<TextField isTitle text={'Test text'} type={SMALL_TYPE} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render text component with text and title and fontSize of 16px', () => {
        const tree = create(<TextField isTitle text={'Test text'} type={EXTRA_TYPE} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render text component with text and title and fontSize of 20px', () => {
        const tree = create(<TextField isTitle text={'Test text'} type={BIG_TYPE} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render text component with text and title and fontSize of 32px', () => {
        const tree = create(<TextField isTitle text={'Test text'} type={HIGH_TYPE} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});