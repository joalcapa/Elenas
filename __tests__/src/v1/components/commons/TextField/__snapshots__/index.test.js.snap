// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component render empty text component 1`] = `
<div
  className="css-text-901oao"
  dir="auto"
  onClick={[Function]}
  style={
    Object {
      "fontSize": "14px",
    }
  }
>
  
</div>
`;

exports[`TextField component render text component with text 1`] = `
<div
  className="css-text-901oao"
  dir="auto"
  onClick={[Function]}
  style={
    Object {
      "fontSize": "14px",
    }
  }
>
  Test text
</div>
`;

exports[`TextField component render text component with text and title 1`] = `
<div
  className="css-text-901oao r-fontWeight-vw2c0b"
  dir="auto"
  onClick={[Function]}
  style={
    Object {
      "fontSize": "14px",
    }
  }
>
  Test text
</div>
`;

exports[`TextField component render text component with text and title and fontSize of 12px 1`] = `
<div
  className="css-text-901oao r-fontWeight-vw2c0b"
  dir="auto"
  onClick={[Function]}
  style={
    Object {
      "fontSize": "12px",
    }
  }
>
  Test text
</div>
`;

exports[`TextField component render text component with text and title and fontSize of 14px 1`] = `
<div
  className="css-text-901oao r-fontWeight-vw2c0b"
  dir="auto"
  onClick={[Function]}
  style={
    Object {
      "fontSize": "14px",
    }
  }
>
  Test text
</div>
`;

exports[`TextField component render text component with text and title and fontSize of 16px 1`] = `
<div
  className="css-text-901oao r-fontWeight-vw2c0b"
  dir="auto"
  onClick={[Function]}
  style={
    Object {
      "fontSize": "16px",
    }
  }
>
  Test text
</div>
`;

exports[`TextField component render text component with text and title and fontSize of 20px 1`] = `
<div
  className="css-text-901oao r-fontWeight-vw2c0b"
  dir="auto"
  onClick={[Function]}
  style={
    Object {
      "fontSize": "20px",
    }
  }
>
  Test text
</div>
`;

exports[`TextField component render text component with text and title and fontSize of 32px 1`] = `
<div
  className="css-text-901oao r-fontWeight-vw2c0b"
  dir="auto"
  onClick={[Function]}
  style={
    Object {
      "fontSize": "32px",
    }
  }
>
  Test text
</div>
`;
